languagetool-core/src/main/java/org/languagetool/tokenizers/WordTokenizer.java

```diff
@@ -217,5 +218,104 @@ private boolean urlEndsAt(int i, List<String> l, String urlQuote) {
     }
     return false;
   }
+  /** This function is designed to handle words like 12.3.a .   
+   * 
+   * @param tokens
+   * @param input
+   * @return
+   */
+  protected List<String> joinValidWord(List<String> tokens, String input) {
+    
+    int count = 0;
+    Matcher matcher = ONE_WORD_TWO_DOT_PATTERN.matcher(input);
+    int noOfMatch = 0;
+    while (matcher.find())
+      noOfMatch++;
+ 
+    if(noOfMatch==0){
+      return tokens;
+    }
+    matcher = ONE_WORD_TWO_DOT_PATTERN.matcher(input);
+    
+    while (matcher.find()) {
+      boolean executed = false;
+      String matchedStr = input.substring(matcher.start() + 1, matcher.end());
+      count++;
+      // Token Is Not At End
+      if(matcher.start()!=0 && input.charAt(matcher.start())!= ' ' && input.charAt(matcher.start())!=','){
+    	  continue;
+      }
+    	  
+      if (matcher.end() != input.length()) {
+        if (input.charAt(matcher.end()) == ',' || input.charAt(matcher.end()) == ' ') {
+
+          if (matcher.start() == 0 && input.charAt(0)!=' ' && input.charAt(0)!=',') {
+            matchedStr = input.substring(matcher.start(), matcher.end());
+          }
+
+          int index = getStartIndex(tokens, matchedStr);
+          // Break The String
+          if (index != -1) {
+
+            List<String> tokenList = tokens.subList(index, index + 5);
+            String newToken = String.join("", tokenList);
+            List<String> frontList = tokens.subList(0, index);
+            frontList.add(newToken);
+            List<String> backList = tokens.subList(index + 6, tokens.size());
+            frontList.addAll(backList);
+            tokens = frontList;
+            executed = true;
+          }
+
+        }
+      }
+
+      if (count == noOfMatch && executed == true)
```

Comment:
Little remark: Comparing a boolean variable to true or false (`executed==true`) is redundant. Either it is true, or it is false.

Reply:
Yeah ,you are right.Thanks for pointing this